### ISSUE TITLE
Clean up the mockLog and spyLog implementations

### ISSUE THREAD

**Issue by philbooth** [mozilla/fxa-auth-server#1655]
There are a couple of funny things about the implementations of `spyLog` and `mockLog`.

Firstly, why do we need two of them? If somebody doesn't need to interrogate what calls were made they can just not bother doing that, we don't need a separate mock object which makes it impossible.

Secondly, `spyLog` uses `sinon` but also contains a home-baked facsimile of some of the functionality from `sinon`. Maybe I'm guilty of not reading the code accurately, but it looks like that home-baked implementation can be ditched.

So, to summarise, I suggest we remove `mockLog`, then rename `spyLog` to `mockLog`, then remove the home-baked spy functionality and just lean on `sinon` instead.

**Comment by rfk** [comment]
> spyLog uses sinon but also contains a home-baked facsimile of
> some of the functionality from sinon.

This probably stems from me not really understanding sinon's capabilities when I was building out this class initially, I'm a big 👍 to what you suggest here.
